apps/server/src/modules/files-storage/repo/fileRecordDO.mapper.ts

```diff
@@ -0,0 +1,86 @@
+import { InternalServerErrorException } from '@nestjs/common';
+import { BaseDOProps } from '@shared/domain';
+import { FileRecord, IFileRecordParams } from '../domain';
+import { FileRecordEntity } from './filerecord.entity';
+
+// TODO: change naming Entity to Datarecord, or without entity
+// TODO: Add interface / Base class that all mappers include the same structure? Naming when no entity is avaible?
+// TODO: include factory methods..
+export class FileRecordDOMapper {
+	static entityToDO(fileRecordEntity: FileRecordEntity): FileRecord {
+		const props: IFileRecordParams & BaseDOProps = {
+			id: fileRecordEntity.id,
+			size: fileRecordEntity.size,
+			name: fileRecordEntity.name,
+			mimeType: fileRecordEntity.mimeType,
+			parentType: fileRecordEntity.parentType,
+			parentId: fileRecordEntity.parentId,
+			schoolId: fileRecordEntity.schoolId,
+			creatorId: fileRecordEntity.creatorId,
+			securityCheck: fileRecordEntity.securityCheck,
+			deletedSince: fileRecordEntity.deletedSince,
+		};
+
+		const fileRecordDO = new FileRecord(props);
+
+		return fileRecordDO;
+	}
+
+	static entitiesToDOs(fileRecordEntities: FileRecordEntity[]): FileRecord[] {
+		const fileRecordDOs = fileRecordEntities.map((fileRecordEntity) => FileRecordDOMapper.entityToDO(fileRecordEntity));
+
+		return fileRecordDOs;
+	}
+
+	/* DO NOT WORK
+	static getEntitiesFromDOs(fileRecords: FileRecord[]): FileRecordEntity[] {
+		const props = fileRecords.map((fileRecord) => fileRecord.getProps());
+		const entities = FileRecordDOMapper.createNewEntities(props);
+
+		return entities;
+	}
+	*/
+
+	static mergeDOsIntoEntities(fileRecords: FileRecord[], fileRecordEntities: FileRecordEntity[]) {
+		fileRecords.forEach((fileRecord) => {
+			const fileRecordEntity = fileRecordEntities.find((entity) => entity.id === fileRecord.id);
+			if (fileRecordEntity) {
+				FileRecordDOMapper.mergeDOintoEntity(fileRecord, fileRecordEntity);
+			} else {
+				throw new InternalServerErrorException('FileRecordDOMapper: ID not found.');
+			}
+		});
+	}
+
+	// For collection the ORM setter must be used.
+	// For embedded entities it must implement to, but should be in a additional method? additional const before
+	// On this place it is possible to match DOs that only include some of the entity keys.
+	// We must make the descision if all keys of the DO should be added or only the possible updates for now?
+	// Should no default values are added on this place?
+	static mergeDOintoEntity(fileRecord: FileRecord, fileRecordEntity: FileRecordEntity) {
```

Comment:
Why do we have to merge a domain object into an entity? We could just map the domain object to a persistence entity (with same id) and then use:
- `em.assign(existingEntity, entity)` for update
- `em.create(EntityClass, entity, { managed: true, persist: true })`for create
and then `em.flush()` to let the entity manager detect and persist the changes.

example: https://github.com/hpi-schul-cloud/schulcloud-server/blob/076aa02e19879656159b252167510a78b7779bbc/apps/server/src/modules/board/repo/board-node.repo.ts#L38-L56